src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3LinuxDriver.cs

```diff
@@ -113,9 +115,14 @@ protected internal override void AddCallbackForPinValueChangedEvent(int pinNumbe
         {
             ValidatePinNumber(pinNumber);
 
-            _interruptDriver.OpenPin(pinNumber);
-            _pinModes[pinNumber].InUseByInterruptDriver = true;
+            _interruptDriver!.OpenPin(pinNumber);
+
+            if (_pinModes[pinNumber] is null)

```

Comment:
I think that because of my comment above of the GpioController already guarding against this, it is probably safe for us to remove this null check since we are already checking for it higher up in the stack.